### DISCUSSION TITLE
GitHub Package Registry (Ruby) Support

### DISCUSSION THREAD

**Discussion by gencer** [renovatebot/renovate#10112] [Request Help]
Does Renovate supports GitHub Package Registry preferably Ruby `bundler`? If so what would be the basic `renovate.json` implementation to support this?

I am asking this because I am unable to install packages from GitHub Package Registry. It gives 404 errors for packages...

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "ignorePresets": [
    ":prHourlyLimit2"
  ],
  "bundler": {
    "registryUrls": [
      "https://rubygems.pkg.github.com/MY-ORG/"
    ]
  },
  "hostRules": [
    {
      "matchHost": "https://rubygems.pkg.github.com/MY-ORG/",
      "encrypted": {
        "token": "***EncryptedToken***"
      }
    }
  ]
}
```

Log file says:

```
DEBUG: Datasource 404
{
  "datasource": "rubygems",
  "lookupName": "sequel-bulk-audit",
  "url": "https://rubygems.pkg.github.com/MY-ORG/api/v1/gems/sequel-bulk-audit.json"
}
DEBUG: Failed to look up dependency sequel-bulk-audit (sequel-bulk-audit)(packageFile="Gemfile", dependency="sequel-bulk-audit")
```

This log belongs to `#360383104`

I'm using GitHub App.

I've tried almost every possible combination but my private gems are not found. (`matchHost`, `baseUrl`, `baseDomain`, `hostType`, `username and password`, `encrypted token`)

@rarkins Maybe you have more info then me or anyone else on this matter?

**Comment by rarkins** [comment]
Have you checked if the URL is right? e.g. using `curl` or equivalent, you can fetch the JSON from the URL in the error? I think it's possible we may need a code change if Rubygems v1 API is not supported and therefore the URL is wrong - not sure anyone's tried it before.

**Reply by gencer** [reply]
Haven't checked yet but probably v1 api is not supported. Will check now

**Reply by gencer** [reply]
@rarkins nope. 404 on my end too:

```
HTTP/2 404
content-type: text/plain; charset=utf-8
x-content-type-options: nosniff
date: Sat, 22 May 2021 14:41:51 GMT
content-length: 19
x-github-request-id: CF23:4877:5FBA8:AADB7:60A9182F

404 page not found
```

It seems this endpoint is not supported by GitHub Packages.

**Reply by rarkins** [reply]
In that case can you please raise a feature request issue to support GitHub Package Registry for Ruby? Ideally with any URLs you can find where GitHub describes what's supported and what's not.

**Reply by tetienne** [reply]
It looks like this issue is already reported [here](https://github.com/renovatebot/renovate/issues/15873)
